### PR TITLE
Polish 2.0

### PR DESCRIPTION
**Pull request by jsiems**
Polishing part 2

### CODE REVIEW

File: lib/Ama/Model/Questions.pm [hunk 1]
```diff
@@ -90,7 +90,7 @@ sub getQuestions {
   }
   
   if($keyword ne 'none') {
-    $sql = $sql . "question like '%" . $keyword . "%' and ";
+    $sql = $sql . "LOWER(question) like LOWER('%" . $keyword . "%') and ";
```
Comment (s1037989) [inline lib/Ama/Model/Questions.pm:93]: This looks like an SQL injection point.  Using DBI's ? feature for auto-quoting.

**Comment by s1037989** [comment]
Aside from the SQL injection of $keyword, :+1:

**Comment by jsiems** [comment]
Ben and I tried for a very long time to inject sql there, and for some reason it never worked.

From what I understand, it SHOULD be vulnerable to sql injections at that point but we were never able to actually get anything to work. I tried for a very long time also to insert some sql (about an hour one day) and never got it. 

I _think_ that it is safe from the sql injection for a reason I do not know. 

Ask Ben about it next time you see him because he knows more about what is going on with it than me(hopefully). Very strange behavior.

**Comment by s1037989** [comment]
Awesome!  Glad to know you tested it!  And with perseverance!!

**Comment by s1037989** [comment]
It looks like it has to do with [L74](https://github.com/KeystoneIT/Ama/blob/master/lib/Ama.pm#L74) of Ama.pm.  It seems your search keywords are passed in the URL, like this:

https://c9.io/questions/:creator/:answered/:orderby/:direction/:limit/:keyword. where :keyword might be snacks+in+the+kitchen, or ;DELETE+FROM+users; and you can see how that wouldn't translate properly.  So, a positive byproduct is the safety from SQL injections; but from experience, I can tell you that passing queries as part of the URL path is not at all normal.  I can't necessarily tell you why...  Perhaps the only reason is the 255 character URL path limit?  Or the 4096 character URL limit?  What if you wanted to allow a query that's 5000 characters?  Now, you might specifically not want to, but that doesn't make this approach acceptable as a result.

I would suggest that all of those filter options should be passed as JSON data, not as RESTful API components.  You might look at [Swagger2](http://search.cpan.org/~jhthorsen/Swagger2-0.83/lib/Mojolicious/Plugin/Swagger2.pm) or [OpenAPI](https://github.com/jhthorsen/mojolicious-plugin-openapi).

**Comment by meyerben** [comment]
That's exactly what I found when I was testing it. I imagine there may be someone creative enough to write up a query that could achieve something, but I haven't been able to find any examples online.

Maybe a new issue for implementing a better filter functionality could be a good idea?

**Pull request merged**